internal/build/imgsrc/resolver.go

```diff
@@ -145,6 +146,7 @@ func (di DeploymentImage) ToSpanAttributes() []attribute.KeyValue {
 type Resolver struct {
 	dockerFactory *dockerClientFactory
 	apiClient     flyutil.Client
+	heartbeatFn   func(ctx context.Context, client *dockerclient.Client, req *http.Request) error
```

Comment:
Paused here for a moment on the use of `ctx` vs `req.Context()` but I'm guessing we want to use ctx on purpose. LGTM